### PR TITLE
Build js in docker image

### PR DESCRIPTION
**Pull request by davkal**

### CODE REVIEW

**Comment by davkal** [comment]
Build runs through:

```
...
Removing intermediate container f0c6c3d7910e
Step 5 : ADD . /app
 ---> 6c38dbc573c8
Removing intermediate container 25d4baff620b
Step 6 : CMD gulp
 ---> Running in 564f1fc4daaa
 ---> 09f837119735
Removing intermediate container 564f1fc4daaa
Successfully built 09f837119735
```

but there is no `dist` dir :(

```
$ docker run weave/scopeui ls
Dockerfile
README.md
app
gulpfile.js
node_modules
package.json
test
```

@tomwilkie any idea?

File: client/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+# build tool
+RUN npm install -g gulp
+
+# install app and build dependencies
+ADD package.json /app/
+RUN npm install
+
+# source files are current dir
+ADD . /app
```
Comment (tomwilkie) [inline client/Dockerfile]: Don't include the src in the docker image.  Expect people to bind mount it in.

**Comment by tomwilkie** [comment]
Try running with 
    docker run -v ./app:/app weave/scopeui

**Comment by davkal** [comment]
@tomwilkie relative paths dont fly. But this line works:
`docker run -v $GOPATH/src/github.com/weaveworks/scope/client:/app weaveworks/scope-build`

Over to you.

**Comment by tomwilkie** [comment]
This fixes #12

**Comment by tomwilkie** [comment]
@davkal this is GTG; PTAL and LGTM!

**Comment by davkal** [comment]
Works great!

* I got into trouble when I had an old `client/dist` dir lying around, the JS was not built then
* maybe depend on `client/dist/scripts/bundle.js`, `client/dist/styles/main.css` and `client/dist/index.html` instead

**Pull request merged**